packages/config/src/Devices.ts

```diff
@@ -41,66 +41,107 @@ export const devicesDir = path.join(configDir, "devices");
 export const indexPath = path.join(devicesDir, "index.json");
 export type DeviceConfigIndex = DeviceConfigIndexEntry[];
 
-/** @internal */
-export async function loadDeviceIndexInternal(): Promise<DeviceConfigIndex> {
-	if (!(await pathExists(indexPath))) {
-		return generateDeviceIndex();
-	}
-	try {
-		const fileContents = await readFile(indexPath, "utf8");
-		return JSON5.parse(fileContents);
-	} catch (e: unknown) {
-		if (e instanceof ZWaveError) {
-			throw e;
-		} else {
-			throwInvalidConfig("devices");
+async function hasChangedDeviceFiles(
+	dir: string,
+	lastChange: Date,
+): Promise<boolean> {
+	// Check if there are any files BUT index.json that were changed
+	// or directories that were modified
+	const filesAndDirs = await fs.readdir(dir);
+	for (const f of filesAndDirs) {
+		const fullPath = path.join(dir, f);
+
+		const stat = await fs.stat(fullPath);
+		if (
+			(dir !== devicesDir || f !== "index.json") &&
+			(stat.isFile() || stat.isDirectory()) &&
+			stat.mtime > lastChange
+		) {
+			return true;
+		} else if (stat.isDirectory()) {
+			// we need to go deeper!
+			if (await hasChangedDeviceFiles(fullPath, lastChange)) return true;
```

Comment:
```suggestion
			return hasChangedDeviceFiles(fullPath, lastChange);
```

Reply:
Also is mtime and lastChange in timestamp format? Otherwise I suggest to use getTime to comprare the dates

Reply:
That proposed change is not correct. We loop through the sub-directories (depth first) and if one directory has (deep) changes, we bail. If not, we need to continue with the next sub-directory.
Your change makes it so that we dive into the first directory and ignore all others.

`mtime` is a Date object, so is `lastChange` (see line 46):
```
lastChange: Date,
```

Reply:
Yep sorry, by looking at it again it's wrong :)

Reply:
About dates, I see them are `Date` objects, usually when dealing with dates I prefer to convert them to millis (Date.getTime()) as for example the equality operator would not work between two dates. But it's ok you can keep it like it is :)

Reply:
> the equality operator would not work between two dates

I've stumbled across that, but `>` and `>=` work :)